Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -373,169 +258,156 @@ class HistoryPanel: SiteTableViewController, HomePanel {
         return cell
     }
 
-    func numberOfSections(in tableView: UITableView) -> Int {
-        var count = 1
-        for category in self.categories where category.rows > 0 {
-            count += 1
+    // MARK: - Selector callbacks
+
+    @objc func onNotificationReceived(_ notification: Notification) {
+        reloadData()
+
+        switch notification.name {
+        case .FirefoxAccountChanged, .PrivateDataClearedHistory:
+            if profile.hasSyncableAccount() {
+                resyncHistory()
+            }
+            break
+        case .DynamicFontChanged:
+            if emptyStateOverlayView.superview != nil {
+                emptyStateOverlayView.removeFromSuperview()
+            }
+            emptyStateOverlayView = createEmptyStateOverlayView()
+            resyncHistory()
+            break
+        default:
+            // no need to do anything at all
+            print("Error: Received unexpected notification \(notification.name)")
+            break
         }
-        return count
     }
 
-    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        if indexPath.section == 0 {
-            self.tableView.deselectRow(at: indexPath, animated: true)
-            return indexPath.row == 0 ? self.showRecentlyClosed() : self.showSyncedTabs()
-        }
-        if let site = self.siteForIndexPath(indexPath), let url = URL(string: site.url) {
-            let visitType = VisitType.typed    // Means History, too.
-            if let homePanelDelegate = homePanelDelegate {
-                homePanelDelegate.homePanel(self, didSelectURL: url, visitType: visitType)
-            }
-            return
+    @objc func onLongPressGestureRecognized(_ longPressGestureRecognizer: UILongPressGestureRecognizer) {
+        guard longPressGestureRecognizer.state == .began else { return }
+        let touchPoint = longPressGestureRecognizer.location(in: tableView)
+        guard let indexPath = tableView.indexPathForRow(at: touchPoint) else { return }
+
+        if indexPath.section != 0 {
+            presentContextMenu(for: indexPath)
         }
-        print("Error: No site or no URL when selecting row.")
     }
 
-    func pinTopSite(_ site: Site) {
-        _ = profile.history.addPinnedTopSite(site).value
+    @objc func onRefreshPulled() {
+        refreshControl?.beginRefreshing()
+        resyncHistory()
     }
 
-    func showSyncedTabs() {
-        let nextController = RemoteTabsPanel(profile: profile)
-        nextController.homePanelDelegate = self.homePanelDelegate
-        self.refreshControl?.endRefreshing()
-        self.navigationController?.pushViewController(nextController, animated: true)
+    // MARK: - UITableViewDataSource
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 5
```

Comment:
If using the `enum  Sections` code style, then that enum can have a `static let count = 5`.
`ActivityStreamPanel` line 159 has the example I followed for this style.